core/src/main/scripts/mutationalSignatures.py

```diff
@@ -0,0 +1,562 @@
+    
+    :param file: open yaml file 
+    :return: yaml key-value pairs
+    """
+    yaml_cont = {}
+    for line in file:
+        key, val = line.strip().split(":", 1)
+        yaml_cont[key.strip()] = val.strip()
+    return yaml_cont
+
+
+def read_mutations_meta_file(study_path: str) -> dict:
+    """Find and read mutation data meta file in cBioPortal study folder
+    
+    :param study_path: path to cBioPortal study folder
+    :return: contents of mutations meta file
+    """
+    meta_files = glob.glob(osp.join(study_path, "meta*"))
+    for file in meta_files:
+        with open(file) as f:
+            yaml_cont = load_yaml(f)
+        if yaml_cont.get("stable_id") == "mutations" and yaml_cont.get("datatype") == "MAF":
+            # Correct file, retrieve associated datafile
+            print(f"Mutation meta file found in study: \'{file}\'")
+            return yaml_cont
+    raise FileNotFoundError(f"Mutation meta file not found in study \'{study_path}\'.")
+
+
+def acquire_maf_study_id(study_path: str) -> tuple:
+    """Find meta file in study and retrieve mutations file and study ID
+    
+    :param study_path: path to cBioPortal study
+    :returns: mutation file path, study ID  
+    """
+    # Find meta files in study
+    yaml_cont = read_mutations_meta_file(study_path)
+    mut_file = yaml_cont.get("data_filename")
+    study_id = yaml_cont.get("cancer_study_identifier")
+    mut_file = osp.join(study_path, mut_file)
+    return mut_file, study_id
+
+
+def get_ncbi_build(maf_df: pd.DataFrame) -> str:
+    """Try to get the ncbi build from a MAF file
+    
+    :param maf_df: dataframe containing 'NCBI_Build' column
+    :return: ncbi build
+    """
+    genome = list(maf_df["NCBI_Build"].unique())
+    if len(genome) != 1:
+        raise ValueError(f"There are either no, or multiple ncbi builds present in your MAF file: \'{genome}\'")
+    return implemented_genome(genome[0].lower())
+
+
+def determine_alternative_allele(maf: pd.DataFrame) -> str:
+    """Determine which alternative allele to use from MAF file
+    
+    :param maf: maf dataframe
+    :return: alternative allele to use 
+    """
+    regexp = re.compile(r'^[ATGC]*$')
+    # This is normally advised against due to speed, but we need only one good row.
+    for idx, row in maf.iterrows():  
+        if row["Tumor_Seq_Allele1"] != row["Reference_Allele"] \
+            and regexp.fullmatch(row["Tumor_Seq_Allele1"]):
+            return "Tumor_Seq_Allele1"
+        elif row["Tumor_Seq_Allele2"] != row["Reference_Allele"] \
+            and regexp.fullmatch(row["Tumor_Seq_Allele2"]):
+            return "Tumor_Seq_Allele2"
+    raise ValueError("Alternative Allele in maf file could not be determined. "
+                     "Please use the --alt-allele parameter.")
+
+
+def preprocess_maf(mut_file: str, tmp_dir: str, args: argparse.Namespace) -> tuple:
+    """Preprocess the cBioPortal mutations data file for matrix generation 
```

Comment:
Perhaps slightly more detail on what preprocessing involves, or why it's done

Reply:
implemented